app/src/org/commcare/mediadownload/MissingMediaDownloadHelper.kt

```diff
@@ -72,29 +72,32 @@ object MissingMediaDownloadHelper : TableStateListener, InstallCancelled {
      * Downloads any missing lazy resources, make sure to call this on background thread
      */
     fun downloadAllLazyMedia(): AppInstallStatus {
-        val platform = CommCareApplication.instance().commCarePlatform
-        val global = platform.globalResourceTable
-
-        global.setInstallCancellationChecker(this)
-        startPinnedNotification()
-
-        val lazyResourceIds = global.lazyResourceIds
-        lazyResourceIds.asSequence()
-                .runCatching {
-                    withIndex()
-                            .onEach { incrementProgress(it.index + 1, lazyResourceIds.size) }
-                            .takeWhile { !wasInstallCancelled() }
-                            .map { global.getResource(it.value) }
-                            .filter { isResourceMissing(it) }
-                            .takeWhile { !wasInstallCancelled() }
-                            .onEach { recoverResource(platform, it) }.toList()
-                }.onFailure {
-                    Logger.log(LogTypes.TYPE_MAINTENANCE, "An error occured while lazy downloading a media resource : " + it.message)
-                    return handleRecoverResourceFailure(it).data
-                }
+        if (!HiddenPreferences.hasLazyMediaDownloaded()) {
```

Comment:
I'm not sure why we'd ever been running lazy media installers for multiple app versions. Are we lazy installing the upgrade table as well as the installed version? or is this due to some behavior that happens after handing over between updates being applied??

Reply:
A new app version can introduce new lazy resources, so once we have successfully run this task for version x, we don't need to run it again for the same version until user updates to a different version. 

Retrospectively, a better change here will be to just maintain a single app pref across all versions and reset that flag after an app update.

Reply:
Ah, got it. I was reviewing by commit, which was silly, and I missed this was old. Sorry for the noise.